routes/api/users.js

```diff
@@ -0,0 +1,209 @@
+        next();
+    }
+});
+
+router.post("/login", async (req, res, next) => {
+    try {
+        const body = req.body;
+        const { error } = userSchema.validate(body);
+
+        if (error) {
+            const validatingErrorMessage = error.details[0].message;
+            return res
+                .status(400)
+                .json({ message: `${validatingErrorMessage}` });
+        }
+
+        const user = await getUserByEmail(body.email);
+
+        if (!user) {
+            return res
+                .status(401)
+                .json({ message: `Email or password is wrong` });
+        }
+
+        const validPassword = await bcrypt.compare(body.password, user.password);
+
+        if (!validPassword) {
+            return res
+                .status(401)
+                .json({ message: `Email or password is wrong` });
+        }
+
+        const token = jwt.sign({ id: user._id }, process.env.JWT_SECRET, {
+            expiresIn: "1y",
+        });
+
+        user.token = token;
+        await user.save();
+
+        res.json({
+            token: `${user.token}`,
+            user: {
+                email: `${user.email}`,
+                subscription: `${user.subscription}`,
+            },
+        });
+        console.log("User login successfully");
+        console.log("User token: ", user.token);
+    } catch (error) {
+        console.error("Error during login: ", error);
+        next();
+    }
+});
+
+router.get("/logout", authenticateToken, async (req, res, next) => {
+    try {
+        const user = req.user;
+
+        if (!user || !user.token) {
+            return res.status(401).json({ message: `Not authorized` });
+        }
+
+        user.token = null;
+        await user.save();
+        res.status(204).json({});
+        console.log("User logout successfully");
+        console.log("User token: ", user.token);
+    } catch (error) {
+        console.error("Error during logout: ", error);
+        next();
+    }
+});
+
+router.get("/current", [authenticateToken, userLoggedIn], async (req, res, next) => {
+    try {
+        const user = req.user;
+        res.json({
+            email: `${user.email}`,
+            subscription: `${user.subscription}`,
+            avatarUrl: `${user.avatarUrl}`,
+        });
+        
+        console.log("User token: ", user.token);
+    } catch (error) {
+        console.error("Something went wrong: ", error);
+        next();
+    }
+});
+
+router.patch(
+    "/avatars",
+    [authenticateToken, userLoggedIn, upload.single("avatar")],
+    async (req, res, next) => {
+    try {
+        const user = req.user;
+        const file = req.file;
+        // const { file, user } = request;
+
+        const avatarUrl = `avatars/${user.id}-${Date.now()}-${file.originalname}`
+            .toLowerCase()
+            .replaceAll(" ", "-");
+
+        await jimp
+            .read(fs.readFileSync(file.path))
+            .then((lenna) => {
+            return lenna
```

Comment:
Staraj się uzywać async await tylko tam gdzie to mozliwe, chodzi o utrzymanie tego samego stylu pisania.